yt_dlp/extractor/zdf.py

```diff
@@ -393,6 +393,7 @@ class ZDFIE(ZDFBaseIE):
             'upload_date': '20250203',
             '_old_archive_ids': ['zdf 240319_2310_sendung_not'],
         },
+        'params': {'skip_download': 'geo-restricted m3u8'},
```

Comment:
This must actually be a geo-restricted HTTP download. The m3u8 formats do not enforce geo-restrictions. Not entirely sure why the HTTP download is preferred here, on the command line it prefers the m3u8 format. Do tests have their own format preference?

Reply:
You're right, it was an http format.

> Do tests have their own format preference?

Yeah, but I can't recall where in the code this happens and it's not *really* relevant to this PR so I can't be bothered to find it, sorry